scripts/test_functions.R

```diff
@@ -0,0 +1,89 @@
+######################################################################
+# Test functions
+######################################################################
+#
+# miscelanous functions for application and database testing
+#
+######################################################################
+
+
+get_db_tables <- function(conn) {
+    # read all database tables and return a list with dataframes
+
+    dbtbls <- list(measurements = tbl(conn,"measurements") %>% collect() ,
+                   sensor = tbl(conn, "location") %>% collect(),
+                   meta = tbl(conn, "meta") %>% collect(),
+                   cache = tbl(conn, "cache") %>% collect()
+    )
+
+    return(dbtbls)
+
+}
+
+
+
+get_rnd_station <- function(dbtables) {
+    # this function creates a randowm set of a station and time range,
+    # this can be used to test the download function
+    #
+    # the input of this function (dbtables) is the output of the
+    # get_db_tables function
+
+    stations <- dbtables$measurements %>% 
+        filter(!str_detect(station, "KNMI"), !str_detect(station, "NL")) %>%
+        distinct(station) %>%
+        pull
+    s <- sample(stations, size = 1)
+
+    last_day <- dbtables$measurements %>%
+        filter(station == s) %>%
+        select(timestamp) %>%
+        max() %>%
+        as_datetime()
+    last_day
+
+    first_day <- dbtables$measurements %>%
+        filter(station == s) %>%
+        select(timestamp) %>%
+        min() %>%
+        as_datetime()
+    first_day
+
+
+    time_start <- last_day
+    time_end <- last_day + days(round(runif(1, min = 1, max = 5)))
+    time_range <- round_to_days(time_start, time_end)
+    time_range
+    res <- list(station = s,
+                first_day = first_day,
+                last_day = last_day,
+                time_start = time_range[1],
+                time_end = time_range[2])
+    return(res)
+}
+
+
+station_overview <- function(conn) {
```

Comment:
Coool :)

Reply:
When we have a few hundredths of stations, we print it out at A0 size, hang it at the office and call it art :-)